volsupervisor/volsupervisor.go

```diff
@@ -42,8 +46,33 @@ retry:
 	go dc.watchAndSetGlobal(globalChan)
 	go info.HandleDebugSignal()
 
+	stopChan, err := lock.NewDriver(dc.Config).AcquireWithTTLRefresh(&config.UseVolsupervisor{Hostname: dc.Hostname}, dc.Global.TTL, dc.Global.Timeout)
+	if err != nil {
+		logrus.Fatal("Could not start volsupervisor: already in use")
+	}
+
+	sigChan := make(chan os.Signal, 1)
+
+	go func() {
+		<-sigChan
+		logrus.Infof("Removing volsupervisor global lock; waiting %v for lock to clear", dc.Global.TTL)
+		stopChan <- struct{}{}
+		time.Sleep(dc.Global.TTL + 1*time.Second) // give us enough time to try to clear the lock
```

Comment:
yeah, derp.

On 22 Aug 2016, at 14:00, Bill Robinson wrote:

>> @@ -42,8 +46,33 @@ retry:
>>  	go dc.watchAndSetGlobal(globalChan)
>>  	go info.HandleDebugSignal()
>>
>> +	stopChan, err := 
>> lock.NewDriver(dc.Config).AcquireWithTTLRefresh(&config.UseVolsupervisor{Hostname: 
>> dc.Hostname}, dc.Global.TTL, dc.Global.Timeout)
>> +	if err != nil {
>> +		logrus.Fatal("Could not start volsupervisor: already in use")
>> +	}
>> +
>> +	sigChan := make(chan os.Signal, 1)
>> +
>> +	go func() {
>> +		<-sigChan
>> +		logrus.Infof("Removing volsupervisor global lock; waiting %v for 
>> lock to clear", dc.Global.TTL)
>> +		stopChan <- struct{}{}
>> +		time.Sleep(dc.Global.TTL + 1*time.Second) // give us enough time 
>> to try to clear the lock
>
> IIRC, we moved away from using `1 * someDuration` in favor of just the 
> bare duration.
>
> ```go
> time.Sleep(dc.Global.TTL + time.Second)
> ```
>
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/contiv/volplugin/pull/435/files/c7d8dfbeaadd372ad97222a731b97939c5c5d542..e1d8c9c912325bd3cc9499e2277ec812df753329#r75757746